File: frontend/src/app/models/order.ts
import { Location } from './location'
import { Warehouse } from './warehouse';

export class Order {
    id: number;
    customer: number;
    retailer: number;
    date_available: string;
    date_delivery: string;
    weight: number;
    ordertimelocation: Location[];
    description: string;
    state: string;
    warehouse: number;

    constructor(customer: number, 
        retailer: number,
        date_available: string,
        weight: number,
        ordertimelocation: Location[],
        description: string,
        warehouse: number,
        state?: string,
        date_delivery?: string,
        ){
            this.customer = customer,
            this.retailer = retailer,
            this.date_available = date_available,
            this.date_delivery = date_delivery,
            this.weight = weight,
            this.ordertimelocation = ordertimelocation ,
            this.description = description,
            this.state = state,
            this.warehouse = warehouse
        }
}

File: backend/delivery/models/order.py
from django.db import models
from django.conf import settings
from .route import Route
from .warehouse import Warehouse
import datetime
from django.utils.translation import gettext_lazy as _

class Order(models.Model):

    class State(models.TextChoices):
        PROCESSING = 'PR', _('Processing')
        READYCTM = 'RC', _('Ready for Custumize')
        CUSTUMIZED = 'CS', _('Custumized')
        READYDIS = 'RD', _('Ready for Distribution')
        DISTRIBUTION = 'DT', _('Distribution')
        COMPLETE = 'CP', _('Complete')

    customer = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name='customer')
    retailer = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name='retailer')
    date_publish = models.DateField(default=datetime.date.today)
    date_available = models.DateField()
    weight = models.DecimalField(decimal_places=2, max_digits=5)
    description = models.CharField(max_length=500, null=True)
    route = models.ForeignKey(Route, related_name='orders', related_query_name="orders", null=True, on_delete=models.CASCADE)
    state = models.CharField(max_length=2, choices=State.choices, default=State.PROCESSING)
    warehouse = models.ForeignKey(Warehouse, related_name='warehouse', null=True, on_delete=models.CASCADE)